pydantic/_internal/_dataclasses.py

```diff
@@ -143,3 +173,41 @@ class B(A):
 
 def is_pydantic_dataclass(_cls: type[Any]) -> TypeGuard[type[PydanticDataclass]]:
     return dataclasses.is_dataclass(_cls) and hasattr(_cls, '__pydantic_validator__')
+
+
+def rebuild_dataclass(
+    cls: type[PydanticDataclass],
+    *,
+    force: bool = False,
+    raise_errors: bool = True,
+    _parent_namespace_depth: int = 2,
+    _types_namespace: dict[str, Any] | None = None,
+) -> bool | None:
+    """
+    Compare with BaseModel.model_rebuild
+
+    TODO: Update this docstring
+    """
+    if not force and cls.__pydantic_complete__:
+        return None
+    else:
+        if _types_namespace is not None:
+            types_namespace: dict[str, Any] | None = _types_namespace.copy()
+        else:
+            if _parent_namespace_depth > 0:
+                frame_parent_ns = _typing_extra.parent_frame_namespace(parent_depth=_parent_namespace_depth) or {}
+                # cls_parent_ns = cls.__pydantic_parent_namespace__ or {}
+                # cls.__pydantic_parent_namespace__ = {**cls_parent_ns, **frame_parent_ns}
```

Comment:
Not sure if we want to store the original namespace on the dataclasses like we do for BaseModel.

This helps in the scenario where rebuilds are necessary when importing from a separate module and there was a ForwardRef to a type in the module in which the class was _originally_ defined.

This may also be necessary when parametrizing generic cyclically-recursive dataclasses outside the module in which they were defined.

I've left these lines present and commented as a reminder to decide about this. I'm okay either way.

Reply:
I'm pretty neutral / would defer to you on it.

My preference is "simple wins", so I think I'd prefer not to store the namespace, but don't really mind.

Reply:
Okay, given there is a lot of complexity related to generic dataclasses and how exactly we should handle them, and not handling it properly is not a regression from v1 (as it wasn't handled properly there), my new plan is to not fiddle with the parent namespace, and also not support generic dataclasses in v2 (but perhaps in v2.1, or _maybe_ sooner if we happen to get it all working in time.. but the pressure is off.)